Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp

```diff
@@ -584,22 +845,58 @@ void XMLHttpRequest::set_onreadystatechange(WebIDL::CallbackType* value)
     set_event_handler_attribute(Web::XHR::EventNames::readystatechange, value);
 }
 
-// https://xhr.spec.whatwg.org/#the-getallresponseheaders()-method
-DeprecatedString XMLHttpRequest::get_all_response_headers() const
+// https://xhr.spec.whatwg.org/#dom-xmlhttprequest-getresponseheader
+WebIDL::ExceptionOr<DeprecatedString> XMLHttpRequest::get_response_header(DeprecatedString const& name) const
 {
-    // FIXME: Implement the spec-compliant sort order.
+    // The getResponseHeader(name) method steps are to return the result of getting name from this’s response’s header list.
+    auto header_bytes = TRY_OR_THROW_OOM(vm(), m_response->header_list()->get(name.bytes()));
+    return header_bytes.has_value() ? DeprecatedString { header_bytes->span() } : DeprecatedString {};
+}
 
-    StringBuilder builder;
-    auto keys = m_response_headers.keys();
-    quick_sort(keys);
+// https://xhr.spec.whatwg.org/#legacy-uppercased-byte-less-than
+static ErrorOr<bool> is_legacy_uppercased_byte_less_than(ByteBuffer const& a, ByteBuffer const& b)
+{
+    // 1. Let A be a, byte-uppercased.
+    auto uppercased_a = TRY(ByteBuffer::copy(a));
+    Infra::byte_uppercase(uppercased_a);
+
+    // 2. Let B be b, byte-uppercased.
+    auto uppercased_b = TRY(ByteBuffer::copy(b));
+    Infra::byte_uppercase(uppercased_b);
 
-    for (auto& key : keys) {
-        builder.append(key);
-        builder.append(": "sv);
-        builder.append(m_response_headers.get(key).value());
-        builder.append("\r\n"sv);
+    // 3. Return A is byte less than B.
+    return Infra::is_byte_less_than(uppercased_a, uppercased_b);
+}
+
+// https://xhr.spec.whatwg.org/#the-getallresponseheaders()-method
```

Comment:
wrong link, should be `https://xhr.spec.whatwg.org/#dom-xmlhttprequest-getallresponseheaders`